clients/src/main/java/org/apache/kafka/common/requests/JoinGroupRequest.java

```diff
@@ -51,6 +51,7 @@ public String toString() {
     private final short version;
 
     public static final String UNKNOWN_MEMBER_ID = "";
+    public static final String EMPTY_GROUP_INSTANCE_ID = "";
```

Comment:
We still need an empty string field to make sure we could correctly serialize the member metadata. Also checking null for String is not very intuitively in Java compared with Scala Option, so my suggestion is to keep using empty string on client side for now.

Reply:
I think on the broker side we should be using `Option` in scala to be consistent with other fields (see my other comment).

On the client side though, it is true that since we only recently dropped J7, `Optional` is not commonly used elsewhere, and I think having it just as a nullable field is fine (we can, do a universal refactoring on client side using `Optional` in another PR but this does not need to be done in this scope).